plugin.go

```diff
@@ -78,6 +83,73 @@ func (p *Plugin) Exec(client *storage.Client) error {
 
 	p.bucket = client.Bucket(strings.Trim(bname, "/"))
 
+	// If in download mode, call the Download method
+	if p.Config.Download {
+		bname, remainingPath := extractBucketName(p.Config.Source)
+		p.Config.Source = remainingPath
+
+		p.bucket = client.Bucket(strings.Trim(bname, "/"))
+
+		log.Println("Downloading objects ...")
+
+		ctx := context.Background()
+		query := &storage.Query{Prefix: p.Config.Source}
+
+		// List the objects in the specified GCS bucket path
+		it := p.bucket.Objects(ctx, query)
+
+		g := errgroup.Group{}
+
+		for {
+			objAttrs, err := it.Next()
```

Comment:
Move the downloading to a different function

Reply:
Introduced a function to perform downloading